### PR TITLE
Subgraphs for primary issue and secondary issue pools

### PR DESCRIPTION
**Pull request by kallolborah**
@mendesfabio @markusbkoch 
I am creating this PR for two new custom pools that we have implemented. I was asked by @johngrantuk to mark the subgraph changes for review by you.

### CODE REVIEW

**Comment by markusbkoch** [comment]
Thanks, we'll review as soon as possible. In the meantime, have you
deployed the resulting subgraph to the hosted service so we can have a
look at it? If not, you doing it would accelerate the process. Thanks!

**Comment by kallolborah** [comment]
> Thanks, we'll review as soon as possible. In the meantime, have you deployed the resulting subgraph to the hosted service so we can have a look at it? If not, you doing it would accelerate the process. Thanks!
> […](#)

@markusbkoch 
Yes, the subgraph is deployed on the hosted service https://thegraph.com/hosted-service/subgraph/verified-network/balancer

**Review by mendesfabio — CHANGES_REQUESTED** [review]
hey, @kallolborah, thanks for this! I reviewed your code and requested some changes. I'm also changing the base branch to `dev`. Please pull the latest changes and fix the conflicts. Feel free to ask any questions 😃

File: schema.graphql [hunk 1]
```diff
@@ -65,6 +65,45 @@ type Pool @entity {
...
+  primarySubscriptions: [PrimaryIssues!] @derivedFrom(field: "poolId")
+  
+  # SecondaryPool only
+  secondaryOffer: BigDecimal
+  secondaryTrades: [SecondaryTrades!] @derivedFrom(field: "poolId")
+}
+
+type PrimaryIssues @entity {
+  id: ID!
+  poolId: Pool!
```
[the remark this answers is not in the document]
Comment (kallolborah) [reply schema.graphql]: Changed it to 'pool', but I see poolId everywhere else, hope it won't break anything now

File: schema.graphql [hunk 2]
```diff
@@ -65,6 +65,45 @@ type Pool @entity {
...
+  investor: Bytes!
+  price: BigDecimal!
+}
+
+type SecondaryTrades @entity {
+  id: ID!
+  poolId: Pool!
+  security: Bytes!
+  party: Bytes! 
+  counterparty: Bytes!
```
[the remark this answers is not in the document]
Comment (kallolborah) [reply schema.graphql]: Referring to User now for both investor and counterparty

File: schema.graphql [hunk 3]
```diff
@@ -65,6 +65,45 @@ type Pool @entity {
...
+  id: ID!
+  poolId: Pool!
+  security: Bytes!
+  party: Bytes! 
+  counterparty: Bytes! 
+  price: BigDecimal!
+  askprice: BigDecimal! 
+  currency: Bytes! 
+  amount: BigDecimal! 
+  status: Bytes!
```
[the remark this answers is not in the document]
Comment (kallolborah) [reply schema.graphql]: I checked this and status is 'pending' or 'settled', so using Bytes here

File: src/mappings/helpers/misc.ts [hunk 4]
```diff
@@ -142,6 +145,14 @@ export function loadPriceRateProvider(poolId: string, tokenAddress: Address): Pr
...
 }
 
+export function loadPrimarySubscriptions(poolId: string, tokenAddress: Address): PrimaryIssues | null {
+  return PrimaryIssues.load(getPoolTokenId(poolId, tokenAddress));
+}
+
+export function loadSecondaryTrades(poolId: string, tokenAddress: Address): SecondaryTrades | null {
+  return SecondaryTrades.load(getPoolTokenId(poolId, tokenAddress));
+}
+
```
[the remark this answers is not in the document]
Comment (kallolborah) [reply src/mappings/helpers/misc.ts]: I checked this again. The Primary and Secondary Issue pools are 2 token pools (1 security + 1 cash token). No BPT is mined for pool joins or pre-mined also. As I understand, `PoolToken` in this case is the poolId-securityAddress which is returned by `getPoolTokenId()` and what `loadPrimarySubscriptions()` and `loadSecondaryTrades()` are doing is using the same PoolToken ID to load subscriptions collected in Primary issue pools and load trades generated in Secondary issue pools. `PrimaryIssues` and `SecondaryTrades` are defined in the schema. Is there anything wrong with this ?

File: schema.graphql [hunk 5]
```diff
@@ -65,6 +65,45 @@ type Pool @entity {
...
+  secondaryOffer: BigDecimal
+  secondaryTrades: [SecondaryTrades!] @derivedFrom(field: "poolId")
+}
+
+type PrimaryIssues @entity {
+  id: ID!
+  poolId: Pool!
+  security: Bytes!
+  assetIn: Bytes!
+  assetName: String!
```
[the remark this answers is not in the document]
Comment (kallolborah) [reply schema.graphql]: `assetIn`, `security`, `currency` are now referencing `Token`

File: schema.graphql [hunk 6]
```diff
@@ -65,6 +65,45 @@ type Pool @entity {
...
+
+type SecondaryTrades @entity {
+  id: ID!
+  poolId: Pool!
+  security: Bytes!
+  party: Bytes! 
+  counterparty: Bytes! 
+  price: BigDecimal!
+  askprice: BigDecimal! 
+  currency: Bytes!
```
[the remark this answers is not in the document]
Comment (kallolborah) [reply schema.graphql]: `party`, `counterparty` are now referencing `User`

**Pull request merged**